graphql_compiler/tests/test_end_to_end.py

```diff
@@ -217,3 +217,57 @@ def test_non_null_types_pass_validation(self):
         arbitrary_argument_name = 'arbitrary_name'
         for graphql_type, value in type_and_value:
             validate_argument_type(arbitrary_argument_name, graphql_type, value)
+
+    def test_date_deserialization(self):
+        # Invalid month
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('birthday', GraphQLDate, '2014-14-01')
+
+        # Invalid day
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('birthday', GraphQLDate, '2014-02-31')
+
+        # Valid date
+        value = deserialize_json_argument('birthday', GraphQLDate, '2014-02-05')
+        self.assertEqual(datetime.date(2014, 2, 5), value)
+
+    def test_datetime_deserialization(self):
+        # No time provided
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('birth_time', GraphQLDateTime, '2014-02-05')
+
+        # No timezone provided
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('birth_time', GraphQLDateTime, '2014-02-05T03:20:55')
+
+        # Invalid format
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('birth_time', GraphQLDateTime, '2014-02-05 03:20:55Z')
+
+        # Valid datetime
+        value = deserialize_json_argument('birth_time', GraphQLDateTime, '2014-02-05T03:20:55Z')
+        self.assertEqual(datetime.datetime(2014, 2, 5, 3, 20, 55, tzinfo=pytz.utc), value)
+
+    def test_float_deserialization(self):
+        # Invalid string
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('amount', GraphQLFloat, 'sdg')
+
+        # Bool
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            deserialize_json_argument('amount', GraphQLFloat, True)
+
+        # Valid string
+        self.assertEqual(float(5), deserialize_json_argument('amount', GraphQLFloat, '5'))
+
+        # Valid string with comma
```

Comment:
nit: Technically a period, not a comma, at least in this localization. Maybe say "decimal portion" or something less locale-sensitive? :)